test/e2e/create_cluster_test.go

```diff
@@ -232,5 +241,50 @@ func validatePrivateCluster(t *testing.T, ctx context.Context, client crclient.C
 		g.Expect(hostedCluster.Status.ControlPlaneEndpoint.Host).ToNot(ContainSubstring("hypershift.local"))
 	}
 
+	for conditionType, expectedStatus := range expectedHCConditionsStatus(hostedCluster) {
+		condition := meta.FindStatusCondition(hostedCluster.Status.Conditions, string(conditionType))
+		g.Expect(condition).ToNot(BeNil())
+		g.Expect(condition.Status).To(Equal(expectedStatus), "%s condition status dosn't match the expected value", conditionType)
+	}
+
 	e2eutil.EnsureNoCrashingPods(t, ctx, client, hostedCluster)
 }
+
+func expectedHCConditionsStatus(hostedCluster *hyperv1.HostedCluster) map[hyperv1.ConditionType]metav1.ConditionStatus {
+	expectedConditions := map[hyperv1.ConditionType]metav1.ConditionStatus{
+		hyperv1.HostedClusterAvailable:          metav1.ConditionTrue,
+		hyperv1.InfrastructureReady:             metav1.ConditionTrue,
+		hyperv1.KubeAPIServerAvailable:          metav1.ConditionTrue,
+		hyperv1.IgnitionEndpointAvailable:       metav1.ConditionTrue,
+		hyperv1.EtcdAvailable:                   metav1.ConditionTrue,
+		hyperv1.ValidReleaseInfo:                metav1.ConditionTrue,
+		hyperv1.ValidHostedClusterConfiguration: metav1.ConditionTrue,
+		hyperv1.SupportedHostedCluster:          metav1.ConditionTrue,
+		hyperv1.ClusterVersionSucceeding:        metav1.ConditionTrue,
+		hyperv1.ClusterVersionUpgradeable:       metav1.ConditionTrue,
```

Comment:
I'm not sure we can guarantee the presence of ClusterVersionUpgradeable, so we can probably relax the expectation to just be != false